### ISSUE TITLE
这就是用openAI的whisper large v3加GPT-style wrapper打了 个包，然后就做为新模型发布了？a shame to China

### ISSUE THREAD

**Issue by sunnysouthfire** [MoonshotAI/Kimi-Audio#140]
月之暗面，你们真的丢尽了中国人的脸！

> **这就是用 OpenAI 的 Whisper large-v3 加 GPT-style wrapper 打了个包，然后就作为新模型发布了？**
> **Is this just Whisper large-v3 + a GPT-style wrapper, repackaged and published as a “new model”?**

---

## 📄 内容正文（中英文双语）

---

### 🧨 中文揭露版：

我认真读完了 Kimi-Audio 的代码，发现了一个很严重的问题：

Kimi-Audio 所谓的“通用音频基础模型”，实际上就是 **把 OpenAI 的 Whisper large-v3 encoder 原封不动搬进来**，然后套了一个 GPT 风格的解码器，再打包成 Hugging Face 模型，发布出来。

关键证据包括：

* 模型内部仍然保留 `whisper_large_v3` 的结构命名；这就是我用过whisper large v3模型的文件。一模一样。太有才了！！！
* 没有任何 encoder 的结构改动；
* tokenizer、预处理管道都是 Whisper 的；
* 完全没有展示过任何自己 pre-train encoder 的过程；
* 模型大小号称 7B，但 HF 页面写的是 9.77B，含糊不清。

这不是真正意义上的“从头训练的基础模型”，而是**把别人训练好的模型套壳**发布的行为。它甚至没有改名，没有技术创新，仅仅包装和宣传。

我不是反对开源项目，我反对的是把“套壳工程”当作“自主研发成果”来宣传，误导开发者和科研人员浪费时间、资源。

---

### 🚨 English version:

After carefully reviewing the Kimi-Audio codebase, I found a critical issue:

The so-called “universal audio foundation model” is **not trained from scratch**, but simply a repackaging of OpenAI’s **Whisper large-v3 encoder** with a GPT-style decoder wrapper.

Key observations:

* The model literally retains `whisper_large_v3` as the encoder name in the code;
* No structural change to the encoder;
* Tokenizer and preprocessing follow Whisper’s design;
* There is **no documentation or evidence** of pretraining their own encoder;
* Model size is ambiguously stated (7B vs. 9.77B on Hugging Face).

This is **not a truly novel model**, but rather a **wrapper around an existing open-source model**, published under a new name.

I’m not against open-source—I'm against misleading developers by dressing up repackaged models as original foundational work. This wastes time, energy, and trust.

**Comment by sunnysouthfire** [comment]
<img width="603" height="977" alt="Image" src="https://github.com/user-attachments/assets/2bab4548-41f1-4305-b6ea-1dbfa674be1a" />

**Comment by sunnysouthfire** [comment]
Kimi-Audio 用的是 Whisper large-v3 的 encoder，所以它不可能真正实现“流式语音识别”！
🎭 Kimi-Audio 官方宣传（误导成分）
在多个页面，包括 GitHub 和媒体稿中，Kimi-Audio 被描述为：

“支持多种音频任务，如语音识别、音频问答、TTS 和流式转写”。

这句话里的“流式转写”最具欺骗性。

<html>
<body>
<h2 data-start="301" data-end="334">🧠 技术原理真相：<strong data-start="314" data-end="334">Whisper-v3 无法做流式</strong></h2>
<p data-start="336" data-end="361">Whisper 的 encoder 架构特点如下：</p>
<div class="_tableContainer_16hzy_1"><div tabindex="-1" class="_tableWrapper_16hzy_14 group flex w-fit flex-col-reverse">
特征 | 是否适合流式 | 说明
-- | -- | --
使用全局 self-attention | ❌ 否 | 每个时间帧依赖全部历史上下文，不能截断处理
chunk-based 输入不支持 | ❌ 否 | 输入必须一次性提供完整音频，无法实时拼接处理
非因果 attention mask | ❌ 否 | Decoder 也必须看到“全部音频编码结果”后才能推断文本
编码器推理耗时高 | ❌ 否 | GPU 解码约 500ms 起步，根本无法低延迟响应
Whisper 官方无流式模式支持 | ❌ 否 | 官方文档明说：Whisper 是 batch-only, non-streamable

</div></div>
</body>
</html>
❌ Whisper v3 结构天生不支持流式；

❌ Whisper decoder 需完整输入才能解码；

❌ Whisper encoder 在推理时显存占用极大，不适合 chunk 推理；

✅ Kimi-Audio 使用 Whisper-v3 做 encoder（源码中保留 whisper_large_v3 名称）；

那么 —— Kimi-Audio 宣称“支持流式语音识别”，是完全不可能成立的。


“Kimi-Audio 的 encoder 用的是 Whisper large-v3，而 Whisper 的结构决定了它不可能支持流式识别。这是公认的架构限制。
所以他们在项目主页上写‘支持流式语音识别’，是完全误导性的营销话术。”

**Comment by HelloWorldU** [comment]
It is indeed not a base model trained from scratch, but only post trained on qwen2.5

**Comment by HelloWorldU** [comment]
Meanwhile, you can take a look about the class named "PrefixStreamingFlowMatchingDetokenizer", which was used for decoding generated audio_tokens.

**Comment by sunnysouthfire** [comment]
> Meanwhile, you can take a look about the class named "PrefixStreamingFlowMatchingDetokenizer", which was used for decoding generated audio_tokens.


> 🔍 Thanks @HelloWorldU for confirming — the model indeed only "wraps" qwen2.5 + whisper-v3 with post-training & detokenizer tricks.
This explains why it’s impossible to do true streaming inference under that architecture.

**Comment by xxr3376** [comment]
@sunnysouthfire Please provide related links for `🎭 Kimi-Audio 官方宣传（误导成分）`, we don't officially saying any about "流式转写"。

The only official wording is in README.md, `chunk-wise streaming detokenizer` only represent for decoding phrase. It's impossible to do chunk-wise streaming input indeed.
